nemo/collections/tts/models/waveglow.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+import torch
+from omegaconf import MISSING, DictConfig, OmegaConf, open_dict
+
+from nemo.collections.tts.helpers.helpers import waveglow_log_to_tb_func
+from nemo.core.classes import ModelPT
+from nemo.utils import logging
+from nemo.utils.decorators import experimental
+
+
+@dataclass
+class PreprocessorParams:
+    pad_value: float = MISSING
+
+
+@dataclass
+class Preprocessor:
+    cls: str = MISSING
+    params: PreprocessorParams = PreprocessorParams()
+
+
+@dataclass
+class WaveglowConfig:
```

Comment:
Do we really need this abstraction everywhere? They can add unnecessary cognitive load.

Reply:
Nope you don't need to define `dataclasses`, but anything that inherits from `ModelPT` should have the following __init__ signature:
```python
def __init__ (self, cfg: 'omegaconf.DictConfig', trainer: 'pytorch_lightning.Trainer' = None):
```
Since cfg is basically a dictionary, I added the dataclasses along with the following code to make sure that the dictionaries are well structured.
```python
        schema = OmegaConf.structured(WaveglowConfig)
        # ModelPT ensures that cfg is a DictConfig, but do this second check in case ModelPT changes
        if isinstance(cfg, dict):
            cfg = OmegaConf.create(cfg)
        elif not isinstance(cfg, DictConfig):
            raise ValueError(f"cfg was type: {type(cfg)}. Expected either a dict or a DictConfig")
        # Ensure passed cfg is compliant with schema
        OmegaConf.merge(cfg, schema)
```